### ISSUE TITLE
[Bug]: Metamask extension freezes when eth _sign_typedData_ v4 is called in recent version

### ISSUE THREAD

**Issue by ianrowan** [MetaMask/metamask-extension#19826]
### Describe the bug

When clicking on a link that worked prior (within my app and others) which prompts an `eth_sign_typedData_v4` the extension freezes and I cannot clear it our unless I restart my browser.
This issue is causing log in to with the method to be impossible

<img width="364" alt="image" src="https://github.com/MetaMask/metamask-extension/assets/36428666/eda5c5f2-2df2-416d-bacf-6d8a84899bb2">

Code used:
```
const signature = await this._web3.givenProvider.request({
      method: 'eth_signTypedData_v4',
      params: [account.address, JSON.stringify(typedMessage)],
    });
 ```

### Steps to reproduce

1. Attempt to call eth_SignedData_v4

### Error messages or log output

```shell
No console error messages
```

### Version

10.32

### Build type

None

### Browser

Chrome

### Operating system

MacOS

**Comment by steddyman** [comment]
Also see the same messages since yesterday trying to do any sign data operations for the EAS easscan.org contracts.  Was working just fine prior to the update.

**Comment by seaona** [comment]
Thank you for reporting this @ianrowan @steddyman !! We are looking into it. Adding here steps to repro:


https://github.com/MetaMask/metamask-extension/assets/54408225/37240f52-759b-41fa-b5cf-db1ec9876228



**Repro**
1. Select Mainnet
2. Go [here](https://easscan.org/attestation/attestWithSchema/0x3969bb076acfb992af54d51274c5c868641ca5344e1aacd0b1f5e4f80ac0822f)
3. Connect MM Wallet
4. Add any address and message
5. Click Make attestation
6. See MM stays loading indefinetly

**Comment by steddyman** [comment]
Thanks.  No, it crashes every time on Sepolia for me.  That is where I am doing all my development.

Also fails on Mainnet.

**Comment by steddyman** [comment]
I think I have found the issue.  It crashes when this option is enabled.  When it is disabled, signing is working again:
<img width="398" alt="image" src="https://github.com/MetaMask/metamask-extension/assets/212404/984291b9-620b-4aee-9b68-176c96682c7f">

**Comment by AlexHerman1** [comment]
@anaamolnar

**Comment by geeksq** [comment]
> I think I have found the issue. It crashes when this option is enabled. When it is disabled, signing is working again: <img alt="image" width="398" src="https://user-images.githubusercontent.com/212404/250066312-984291b9-620b-4aee-9b68-176c96682c7f.png">

thanks for this! after unchecking the experimental feature, MM is working for me again

**Comment by jacobc-eth** [comment]
I'm also receiving this issue.

Chromium
Lattice hardware wallet
Snapshot.org
Ethereum Mainnet

**Comment by jacobc-eth** [comment]
One other workaround for people who don't want to completely kill your Chrome browser:

You can use the Chrome task manager to close MetaMask, then navigate to the extensions view and hit "Reload" to restart metamask.

**Comment by robwilkes** [comment]
So I was just impacted by this, really painful.
What happened in my instance, I had version 10.33.0 installed according to my extensions page, however the UI looked no different.
It looked like the old version, and most things worked fine, however when you went to sign typed data the extension would hang.

I tried clearing all browser data, all extension data, and event uninstalled and reinstalled the extension, each time it didn't work and each time it had the old UI.

I didn't even realise the UI had changed until I completely trashed my entire Brave installation, deleting:
~/Library/Application Support/BraveSoftware
~/Library/Caches/com.brave.Browser
~/Library/Caches/BraveSoftware
~/Library/Preferences/com.brave.Browser.plist

and re-installing Brave from scratch.

Only then when I reinstalled Metamask did I realise it had a new UI, and after that signing typed data worked fine.

So I think the update somehow trashed the installation with some mixture of new and old html & js.

If you're having this issue and your metamask doesn't have this new look, then perhaps you had the same issue I did.

![image](https://github.com/MetaMask/metamask-extension/assets/4636768/572e369b-3e38-4604-8ac6-076cc5375d30)
